mlflow/fastai.py

```diff
@@ -0,0 +1,340 @@
+        with open(conda_env, "r") as f:
+            conda_env = yaml.safe_load(f)
+    with open(os.path.join(path, conda_env_subpath), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.fastai",
+                        data=model_data_subpath, env=conda_env_subpath)
+    mlflow_model.add_flavor(FLAVOR_NAME, fastai_version=fastai.__version__, data=model_data_subpath)
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def log_model(fastai_learner, artifact_path, conda_env=None, registered_model_name=None, **kwargs):
+    """
+    Log a fastai model as an MLflow artifact for the current run.
+
+    :param fastai_learner: Fastai model (an instance of `fastai.Learner`_) to be saved.
+    :param artifact_path: Run-relative artifact path.
+    :param conda_env: Either a dictionary representation of a Conda environment or the path to a
+                      Conda environment yaml file. If provided, this describes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :func:`get_default_conda_env()`. If ``None``, the default
+                      :func:`get_default_conda_env()` environment is added to the model.
+                      The following is an *example* dictionary representation of a Conda
+                      environment::
+
+                        {
+                            'name': 'mlflow-env',
+                            'channels': ['defaults'],
+                            'dependencies': [
+                                'python=3.7.0',
+                                'fastai=1.0.60',
+                            ]
+                        }
+    :param registered_model_name: Note:: Experimental: This argument may change or be removed in a
+                                  future release without warning. If given, create a model
+                                  version under ``registered_model_name``, also creating a
+                                  registered model if one with the given name does not exist.
+    :param kwargs: kwargs to pass to `fastai.Learner.export`_ method.
+    """
+    Model.log(artifact_path=artifact_path, flavor=mlflow.fastai,
+              registered_model_name=registered_model_name,
+              fastai_learner=fastai_learner, conda_env=conda_env, **kwargs)
+
+
+def _load_model(path):
+    from fastai.basic_train import load_learner
+    abspath = os.path.abspath(path)
+    path, file = os.path.split(abspath)
+    return load_learner(path, file)
+
+
+class _FastaiModelWrapper:
+    def __init__(self, learner):
+        self.learner = learner
+
+    def predict(self, dataframe):
+        test_data = TabularList.from_df(dataframe, cont_names=self.learner.data.cont_names)
+        self.learner.data.add_test(test_data)
+        return self.learner.get_preds(DatasetType.Test)[0].numpy()
```

Comment:
Would you be able to clarify why we take the first element of `get_preds` here? In what cases can [get_preds](https://docs.fast.ai/basic_train.html#get_preds) return multiple elements / would users be surprised if we only output the first one as the model's prediction?

Reply:
The method get_preds returns a tuple of two tensors, both are the predictions but the second tensor contains classes (0, 1, 2, ..., N) in case of classification, while the first one contains the actual probabilities.

https://docs.fast.ai/basic_train.html#Learner.get_preds